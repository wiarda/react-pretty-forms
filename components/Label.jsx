/* eslint-disable react/prop-types */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
const React = require('react');

function Label({
  label,
  labelTextClassName,
  children,
  ...props
}) {
  if (label) {
    return (
      <label
        {...props}
      >
        <span className={labelTextClassName} data-active={props['data-active']}>
          {label}
        </span>
        {children}
      </label>
    );
  }
  return (
    <React.Fragment>
      {children};
    </React.Fragment>
  );
}

module.exports = React.memo(Label);
