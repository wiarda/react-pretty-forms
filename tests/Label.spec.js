/* global describe test expect */
import React from 'react';
import renderer from 'react-test-renderer';
import Label from '../components/Label';

describe('Label component', () => {
  test('Matches snapshot', () => {
    const component = renderer.create(<Label label="Test Label" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders required props', () => {
    const component = renderer
      .create(
        <Label Label="Register">
          <div>Child</div>
        </Label>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Accepts custom styles', () => {
    const component = renderer
      .create(<Label label="button" labelTextClassName="test-class" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
