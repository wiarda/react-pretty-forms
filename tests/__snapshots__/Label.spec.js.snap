// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component Accepts custom styles 1`] = `
<label>
  <span
    className="test-class"
  >
    button
  </span>
</label>
`;

exports[`Label component Matches snapshot 1`] = `
<label>
  <span>
    Test Label
  </span>
</label>
`;

exports[`Label component Renders required props 1`] = `
Array [
  <div>
    Child
  </div>,
  ";",
]
`;
